torch/lib/c10d/NCCLUtils.hpp

```diff
@@ -16,11 +16,25 @@
     if (error != ncclSuccess) {                                           \
       std::string err = "NCCL error in: " + std::string(__FILE__) + ":" + \
           std::to_string(__LINE__) + ", " +                               \
-          std::string(ncclGetErrorString(error));                         \
+          std::string(ncclGetErrorString(error)) + ", NCCL version " +    \
+          getNcclVersion();                                               \
       throw std::runtime_error(err);                                      \
     }                                                                     \
   } while (0)
 
+inline std::string getNcclVersion() {
```

Comment:
I'm not sure what's the best way to define this function `NCCLUtils.hpp`. If I put it in an anonymous namespace, then it won't be available to `ProcessGroupNCCL`, but I need it there. If I don't inline the function the linker gives multiple definition errors. Any suggestions?

Reply:
Without its own translation unit you either have to inline or put it in an anonymous namespace in this header (which is essentially equivalent to inline). You can avoid duplication by adding a `NCCLUtils.cpp` file and adding it in there. If you do so, you can also make this use a static `once_flag` and ensure you only do this once, in case this routine is abused and called a lot some time in the future.